front/app/modules/free/native_surveys/admin/components/SurveyBuilderSettings/index.tsx

```diff
@@ -1,59 +1,157 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 
 // styles
-import styled from 'styled-components';
 import { colors } from 'utils/styleUtils';
 
 // components
-import { Title, Box, stylingConsts } from '@citizenlab/cl2-component-library';
+import {
+  Box,
+  Toggle,
+  Text,
+  stylingConsts,
+} from '@citizenlab/cl2-component-library';
 import Button from 'components/UI/Button';
+import InputMultilocWithLocaleSwitcher from 'components/UI/InputMultilocWithLocaleSwitcher';
+import { SectionField, SectionTitle } from 'components/admin/Section';
 import CloseIconButton from 'components/UI/CloseIconButton';
 
 // intl
-import messages from './messages';
+import messages from '../messages';
 import { FormattedMessage } from 'utils/cl-intl';
 
-const StyledBox = styled(Box)`
-  box-shadow: -2px 0px 1px 0px rgba(0, 0, 0, 0.06);
-`;
-
-const SurveyBuilderSettings = () => (
-  <StyledBox
-    position="fixed"
-    right="0"
-    top={`${stylingConsts.menuHeight}px`}
-    zIndex="99999"
-    p="20px"
-    w="400px"
-    h="100%"
-    background="#ffffff"
-  >
-    <Box position="absolute" right="8px">
-      <CloseIconButton
-        a11y_buttonActionMessage={messages.close}
-        onClick={() => {}}
-        iconColor={colors.label}
-        iconColorOnHover={'#000'}
-      />
-    </Box>
-    <Title variant="h2">
-      <FormattedMessage {...messages.shortAnswer} />
-    </Title>
-    <Box display="flex">
-      <Button
-        icon="delete"
-        buttonStyle="primary-outlined"
-        borderColor={colors.red500}
-        textColor={colors.red500}
-        iconColor={colors.red500}
-        onClick={() => {
-          // TODO: Handle delete
-        }}
-      >
-        <FormattedMessage {...messages.delete} />
-      </Button>
+// Types
+import { Multiloc } from 'typings';
+
+import {
+  IFlatCustomField,
+  IFlatUpdateCustomField,
+} from 'modules/free/native_surveys/services/surveyCustomFields';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+
+interface Props {
+  field?: IFlatCustomField;
+  onDelete: (fieldId: string) => void;
+  onFieldChange: (field: IFlatUpdateCustomField) => void;
+  onClose: () => void;
+}
+
+const SurveyBuilderSettings = ({
+  field,
+  onDelete,
+  onFieldChange,
+  onClose,
+}: Props) => {
+  // I'm keeping this form as simple as possible using state pending form rework from (TEC-35)
+  const [fieldState, setFieldState] = useState({
+    isRequired: field?.required || false,
+    questionTitle: field?.title_multiloc || {},
+    questionDescription: field?.description_multiloc || {},
+  });
+
+  useEffect(() => {
+    if (!isNilOrError(field)) {
+      onFieldChange({
+        ...field,
+        id: field.id,
+        title_multiloc: fieldState.questionTitle as Multiloc,
```

Comment:
I've read that using the "as" keyword in TS isn't preferred (I think since it forces the compiler to consider the type as something other than what is automatically infers?) - do you think we could do this typing another way? Or is this required here?